CMakeLists.txt

```diff
@@ -106,6 +106,7 @@ zephyr_library_named(zephyr)
 
 zephyr_include_directories(
   include
+  include/zephyr
```

Comment:
Should this be conditional to a Kconfig option? That way we can leave a compatibility mode, but also have an option to help migrate downstream projects or ensure that we don't reintroduce old path includes after this.

Reply:
I agree we should have a Kconfig option. Other things I'd consider, given that a future removal of `include/` option will break the compilation of every single file:

- Deprecate current `include/` path, but keep it the default except for upstream code
- Add a big warning when building with the "compatible" mode until it is not deleted (users will notice this way)
- Maybe provide a migration script that automatically prefixes every Zephyr header used in apps/libs with `zephyr/`

Reply:
Yes, I just wanted to get the CI to pass first before I add another layer to it.